### PR TITLE
Feat #65: Improve balance lose check and mention it's details in docs

### LINKED ISSUE #65: MAINNET SOR losing ADA in some cases
**Describe the bug**
 - In some exceptional scenarios the MAINNET SOR seems to be losing minimal amounts of ADA. There was a reported case of the SOR losing 0.5 ADA, which is not much, but it should not happen.

**To Reproduce**
1. Run a MAINNET SOR instance
2. Observe the submitted transactions
3. Observe a tranasaction losing 0.5ADA

**Expected behavior**
The SOR should not be losing ADA.

**Screenshots**
 - N/A

**Additional context**
The issue was reported via Discord:
- https://discord.com/channels/851910000003645490/1185078070856663130/1185930995191529532

Example transactions that lost ADA:
- https://cexplorer.io/tx/c86a92232f9065c884570a01582b0651e9a7b63868b8fbbc247adf5d3e328b86
- https://cexplorer.io/tx/d060323d1a6d550e5972703e26f59973d543187f1dd0abcd499ada185dbdda43

### PR DESCRIPTION
**Pull request by sourabhxyz**
Closes #65. Also see #62.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,6 +128,10 @@ If we want our earnings to be in `tokenA` then the
 commodity must be `tokenB`. So we can buy from the sell order, `20 tokenB` using `8 tokenA`, then
 using these `20 tokenB` we can get `10 tokenA` from the buy order, earning `2 tokenA`.
 
+> **ⓘ **
+>
+> Note that as per current design of bot, there is a check in end to determine whether the bot would loose balance in case it submits the build transaction. This check subtracts for transaction fees, that is, it might happen that bot may loose ADA when submitting the transaction but that would solely be due to transaction fees. If currency is set to ADA, bot may still loose some ADA if the transaction fees is more than found ADA arbitrage.
```
Comment (adacapo21) [inline README.md:133]: > the bot would loose balance

I guess you meant to write "the bot would **lose** balance"

**Review by adacapo21 — CHANGES_REQUESTED** [review]
A small repeated typo

File: README.md (hunk 1 above)
Comment (sourabhxyz) [reply README.md:133]: Yes, thanks for pointing it out @adacapo21!

File: README.md [hunk 2]
```diff
@@ -49,8 +49,8 @@ over the multi-asset order book to obtain a list of matches. The matches are the
 translated into transactions that will be signed and submitted by the bot.
 
 Due to the open and decentralized design of the protocol, anybody can run a Smart Order
-Router instance and collect a share of the fees, thus running a Smart Order Router instance
-is not only contributiung to the further decentralization of the protocol, but it is also
+Router instance and collect the arbitrage opportunities, thus running a Smart Order Router instance
```
Comment (4TT1L4) [inline README.md:52]: > collect the arbitrage opportunities,
I think we should change the wording here. One does not "collect" opportunities.

File: README.md [hunk 3]
```diff
@@ -128,6 +128,13 @@ If we want our earnings to be in `tokenA` then the
 commodity must be `tokenB`. So we can buy from the sell order, `20 tokenB` using `8 tokenA`, then
 using these `20 tokenB` we can get `10 tokenA` from the buy order, earning `2 tokenA`.
 
+> **ⓘ **
```
Comment (4TT1L4) [inline README.md:131]: There is a GitHub specific Markdown extension that allows to place special colored content with icons:
 - https://github.com/orgs/community/discussions/16925

I think we should consider using such a box here. `> [!NOTE]  ` sounds like a good choice.

**Review by 4TT1L4 — CHANGES_REQUESTED** [review]
Please see my comments.

File: README.md (hunk 2 above)
Comment (sourabhxyz) [reply README.md:52]: Sorry, could you elaborate / suggest the required changes?

**Comment by adacapo21** [comment]
> Thanks for taking a look @adacapo21! The PR would be merged as soon as it is reviewed from someone else from team.

You are welcome. Good luck!

**Comment by ajuggler** [comment]
The proposed changes ensure that there is no Ada loss, even when taking fees into account, when the currency is lovelaces; and that when the trade involves a non-lovelace currency then the Ada loss is only due to fees paid.  This seems reasonable to me.

**Review by ajuggler — APPROVED** [review]
Proposed changes ensure that when trade only involves lovelace as currency, there is no Ada loss even when taking fees into account.

**Pull request merged**